packages/runtime/container-runtime/src/opTelemetry.ts

```diff
@@ -0,0 +1,46 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IDeltaManager } from "@fluidframework/container-definitions";
+import {
+    IDocumentMessage,
+    ISequencedDocumentMessage,
+} from "@fluidframework/protocol-definitions";
+import { isSystemMessage } from "@fluidframework/protocol-base";
+import { utf8ByteLength } from "@fluidframework/runtime-utils";
+
+export class OpTracker {
+    private _nonSystemOpCount: number = 0;
+    public get nonSystemOpCount(): number {
+        return this._nonSystemOpCount;
+    }
+
+    private _opsSizeAccumulator: number = 0;
+    public get opsSizeAccumulator(): number {
+        return this._opsSizeAccumulator;
+    }
+
+    public constructor(
+        deltaManager: IDeltaManager<ISequencedDocumentMessage, IDocumentMessage>,
+        disabled: boolean,
+    ) {
+        if (disabled) {
+            return;
+        }
+
+        deltaManager.on("op", (message) => {
+            this._nonSystemOpCount += isSystemMessage(message) ? 0 : 1;
+            const stringContents = typeof message.contents === "string" ?
+                message.contents :
+                JSON.stringify(message.contents);
```

Comment:
This can be costly, and it pretty much is for every message I've tested manually. But AFAIK there is no _current_ way of doing it better. I believe the best way would be to stamp the message with its size at creation time, but that's a longer change involving protocol changes.

What do you think, @vladsud ?

Reply:
Your code above (as it stands) does not need to test type - it's always an object (see top of DeltaManager.processInboundMessage where we unpack it).

I'd suggest one of the following:
1. Add an event in DeltaManager.processInboundMessage before op content parsing.
2. Listen for "push" event on DeltaManager.inbound. This is not the same (and not accurate), as we may summarize halfway through processing ops from inbound queue.

I think # 1 is the right approach. Note that it will take time for these changes to propagate to all clients, so some clients will report nothing for a while, but that's fine with me.

Reply:
How about now? I think this approach is both accurate and doesn't need the cross-layer interface changes wrt to the events the DeltaManager exposes (so we can tackle where the DeltaManager resides in a different effort).

Reply:
Well, the issue with inbound queue is that we may summarize when inbound queue is not empty. So, the stats would be correct in long time horizon (we can measure total op counts and sizes), but we would not be able to correctly say metrics for a particular summary.

This might be a good starting point (as size per op and bandwidth requirement are top interest, and questions are not tied to summaries themselves), but it might be not great metric if we want to understand how many user edits (ops) are in a summary.

I guess what I'm getting to - we can do that, but I'd also continue with adding proper event and recording data correctly per summary (and eventually removing logic tied to inbound queue)

Reply:
Ah, I see you measure counts with "op" handler. I think it's good enough for starters for sure :)
But I'd recommend follow up with extra events and unifying logging under same event - it allows proper per summary metric tracking, which is not possible with this flow.